Upstream-Status: Inappropriate [hack]
--- a/src/plugin.h
+++ b/src/plugin.h
@@ -19,14 +19,10 @@
 		};
 #else
 #define BLUETOOTH_PLUGIN_DEFINE(name, version, priority, init, exit) \
-		extern struct btd_debug_desc __start___debug[] \
-				__attribute__ ((weak, visibility("hidden"))); \
-		extern struct btd_debug_desc __stop___debug[] \
-				__attribute__ ((weak, visibility("hidden"))); \
 		extern struct bluetooth_plugin_desc bluetooth_plugin_desc \
 				__attribute__ ((visibility("default"))); \
 		struct bluetooth_plugin_desc bluetooth_plugin_desc = { \
 			#name, version, priority, init, exit, \
-			__start___debug, __stop___debug \
+			0, 0 \
 		};
 #endif
--- a/src/log.c
+++ b/src/log.c
@@ -3,9 +3,6 @@
 	va_end(ap);
 }
 
-extern struct btd_debug_desc __start___debug[];
-extern struct btd_debug_desc __stop___debug[];
-
 static char **enabled = NULL;
 
 static gboolean is_enabled(struct btd_debug_desc *desc)
@@ -39,10 +36,7 @@
 
 void __btd_toggle_debug(void)
 {
-	struct btd_debug_desc *desc;
-
-	for (desc = __start___debug; desc < __stop___debug; desc++)
-		desc->flags |= BTD_DEBUG_FLAG_PRINT;
+	return;
 }
 
 void __btd_log_init(const char *debug, int detach)
@@ -52,8 +46,6 @@
 	if (debug != NULL)
 		enabled = g_strsplit_set(debug, ":, ", 0);
 
-	__btd_enable_debug(__start___debug, __stop___debug);
-
 	bt_log_open();
 
 	if (!detach)
