Upstream-Status: Pending

diff --git a/arch/riscv/cpu/ads_rv32/boot0.lds b/arch/riscv/cpu/ads_rv32/boot0.lds
index c70ff72e..87d9b5d0 100644
--- a/arch/riscv/cpu/ads_rv32/boot0.lds
+++ b/arch/riscv/cpu/ads_rv32/boot0.lds
@@ -9,7 +9,7 @@ SECTIONS
 
 	.head   :
 	{
-		main/boot0_head.o	(.rodata)
+		KEEP(main/boot0_head.o	(.rodata*))
 	}
 	. = ALIGN(1);
 
diff --git a/arch/riscv/cpu/ads_rv32/fes1.lds b/arch/riscv/cpu/ads_rv32/fes1.lds
index f97630d7..beb17a6a 100644
--- a/arch/riscv/cpu/ads_rv32/fes1.lds
+++ b/arch/riscv/cpu/ads_rv32/fes1.lds
@@ -8,7 +8,7 @@ SECTIONS
 
 	.head   :
 	{
-		main/fes1_head.o	(.rodata)
+		KEEP(main/fes1_head.o	(.rodata*))
 	}
 	.text :
 	{
diff --git a/arch/riscv/cpu/ads_rv32/simpleboot.lds b/arch/riscv/cpu/ads_rv32/simpleboot.lds
index 8a2d77ce..9778fe60 100644
--- a/arch/riscv/cpu/ads_rv32/simpleboot.lds
+++ b/arch/riscv/cpu/ads_rv32/simpleboot.lds
@@ -9,7 +9,7 @@ SECTIONS
 
 	.head   :
 	{
-		main/boot0_head.o	(.rodata)
+		KEEP(main/boot0_head.o	(.rodata*))
 	}
 	. = ALIGN(1);
 
