Subject: [PATCH] x1600: add missing SFC clock definition

diff --git a/arch/mips/boot/dts/ingenic/x1600.dtsi b/arch/mips/boot/dts/ingenic/x1600.dtsi
index f885c3d33..bd4bd7992 100644
--- a/arch/mips/boot/dts/ingenic/x1600.dtsi
+++ b/arch/mips/boot/dts/ingenic/x1600.dtsi
@@ -483,6 +483,9 @@ sfc: sfc@13440000 {
 			reg = <0x13440000 0x10000>;
 			interrupt-parent = <&intc>;
 			interrupts = <IRQ_SFC>;
+
+			clocks = <&cgu X1600_CLK_SFC>;
+			clock-names = "sfc";
 		};
 
 		dma: dma-controller@13420000 {
diff --git a/drivers/clk/ingenic/x1600-cgu.c b/drivers/clk/ingenic/x1600-cgu.c
index 2861dd0b8..b1b8f59a5 100644
--- a/drivers/clk/ingenic/x1600-cgu.c
+++ b/drivers/clk/ingenic/x1600-cgu.c
@@ -348,6 +348,15 @@ static const struct ingenic_cgu_clk_info x1600_cgu_clocks[] = {
 		.gate = { CGU_REG_CLKGR0, 22 },
 	},
 
+	[X1600_CLK_SFC] = {
+		"sfc", CGU_CLK_MUX | CGU_CLK_DIV | CGU_CLK_GATE,
+		.parents = { X1600_CLK_SCLKA, X1600_CLK_MPLL,
+			     X1600_CLK_EPLL, -1 },
+		.mux = { CGU_REG_SFCCDR, 30, 2 },
+		.div = { CGU_REG_SFCCDR, 0, 1, 8, 29, 28, 27 },
+		.gate = { CGU_REG_CLKGR0, 2 },
+	},
+
 	[X1600_CLK_EXCLK_DIV512] = {
 		"exclk_div512", CGU_CLK_FIXDIV,
 		.parents = { X1600_CLK_EXCLK },
diff --git a/include/dt-bindings/clock/ingenic,x1600-cgu.h b/include/dt-bindings/clock/ingenic,x1600-cgu.h
index 3ccf60a37..357210bcf 100644
--- a/include/dt-bindings/clock/ingenic,x1600-cgu.h
+++ b/include/dt-bindings/clock/ingenic,x1600-cgu.h
@@ -54,4 +54,5 @@
 #define X1600_CLK_AES			37
 #define X1600_CLK_I2S0_RX		38
 #define X1600_CLK_I2S0_TX		39
+#define X1600_CLK_SFC			40
 
 #endif /* __DT_BINDINGS_CLOCK_X1600_CGU_H__ */
