Subject: [PATCH] x1600: add missing HWRNG clock definition and driver support

diff --git a/arch/mips/boot/dts/ingenic/x1600.dtsi b/arch/mips/boot/dts/ingenic/x1600.dtsi
index bd4bd7992..a6b6fe7c0 100644
--- a/arch/mips/boot/dts/ingenic/x1600.dtsi
+++ b/arch/mips/boot/dts/ingenic/x1600.dtsi
@@ -339,11 +339,14 @@ spi_slv0: slv@10045000 {
 		};
 
 		dtrng: dtrng@10072000 {
-			compatible = "ingenic,dtrng";
+			compatible = "ingenic,x1600-dtrng";
 			reg = <0x10072000 0x100>;
 			interrupt-parent = <&intc>;
 			interrupts = <IRQ_DTRNG>;
 			status = "disabled";
+
+			clocks = <&cgu X1600_CLK_DTRNG>;
+			clock-names = "dtrng";
 		};
 
 		otg_phy: usb-phy@10007800 {
diff --git a/drivers/char/hw_random/Kconfig b/drivers/char/hw_random/Kconfig
index f0dde77f5..ec237e3cd 100644
--- a/drivers/char/hw_random/Kconfig
+++ b/drivers/char/hw_random/Kconfig
@@ -272,7 +272,7 @@ config HW_RANDOM_INGENIC_RNG
 
 config HW_RANDOM_INGENIC_TRNG
 	tristate "Ingenic True Random Number Generator support"
-	depends on MACH_X1830 || COMPILE_TEST
+	depends on MACH_X1830 || MACH_X1600 || COMPILE_TEST
 	default HW_RANDOM
 	help
 	  This driver provides kernel-side support for the True Random Number Generator
diff --git a/drivers/char/hw_random/ingenic-trng.c b/drivers/char/hw_random/ingenic-trng.c
index 1672320e7..d19ef61eb 100644
--- a/drivers/char/hw_random/ingenic-trng.c
+++ b/drivers/char/hw_random/ingenic-trng.c
@@ -110,6 +110,7 @@ static int ingenic_trng_probe(struct platform_device *pdev)
 
 static const struct of_device_id ingenic_trng_of_match[] = {
 	{ .compatible = "ingenic,x1830-dtrng" },
+	{ .compatible = "ingenic,x1600-dtrng" },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, ingenic_trng_of_match);
diff --git a/drivers/clk/ingenic/x1600-cgu.c b/drivers/clk/ingenic/x1600-cgu.c
index b1b8f59a5..26644aeec 100644
--- a/drivers/clk/ingenic/x1600-cgu.c
+++ b/drivers/clk/ingenic/x1600-cgu.c
@@ -445,6 +445,12 @@ static const struct ingenic_cgu_clk_info x1600_cgu_clocks[] = {
 		.gate = { CGU_REG_CLKGR0, 16 },
 	},
 
+	[X1600_CLK_DTRNG] = {
+		"dtrng", CGU_CLK_GATE,
+		.parents = { X1600_CLK_EXCLK, -1, -1, -1 },
+		.gate = { CGU_REG_CLKGR0, 17 },
+	},
+
 	[X1600_CLK_UART3] = {
 		"uart3", CGU_CLK_GATE,
 		.parents = { X1600_CLK_EXCLK, -1, -1, -1 },
diff --git a/include/dt-bindings/clock/ingenic,x1600-cgu.h b/include/dt-bindings/clock/ingenic,x1600-cgu.h
index 357210bcf..2d0c8fd6c 100644
--- a/include/dt-bindings/clock/ingenic,x1600-cgu.h
+++ b/include/dt-bindings/clock/ingenic,x1600-cgu.h
@@ -55,4 +55,5 @@
 #define X1600_CLK_I2S0_RX		38
 #define X1600_CLK_I2S0_TX		39
 #define X1600_CLK_SFC			40
+#define X1600_CLK_DTRNG		41
 
 #endif /* __DT_BINDINGS_CLOCK_X1600_CGU_H__ */
