Subject: [PATCH] mips: x1600: add AES clock references

diff --git a/arch/mips/boot/dts/ingenic/x1600.dtsi b/arch/mips/boot/dts/ingenic/x1600.dtsi
index af4bc2166..be361b3df 100644
--- a/arch/mips/boot/dts/ingenic/x1600.dtsi
+++ b/arch/mips/boot/dts/ingenic/x1600.dtsi
@@ -460,5 +460,7 @@ aes: aes@13430000 {
 			reg = <0x13430000 0x10000>;
 			interrupt-parent = <&intc>;
 			interrupts = <IRQ_AES>;
+			clocks = <&cgu X1600_CLK_AES>, <&cgu X1600_CLK_PDMA>;
+			clock-names = "gate_aes", "gate_pdma";
 			status = "okay";
 		};
diff --git a/drivers/clk/ingenic/x1600-cgu.c b/drivers/clk/ingenic/x1600-cgu.c
index 2861dd0b8..5fe207c1e 100644
--- a/drivers/clk/ingenic/x1600-cgu.c
+++ b/drivers/clk/ingenic/x1600-cgu.c
@@ -451,7 +451,7 @@ static const struct ingenic_cgu_clk_info x1600_cgu_clocks[] = {
 	[X1600_CLK_AES] = {
 		"aes", CGU_CLK_GATE,
 		.parents = { X1600_CLK_AHB2, -1, -1, -1 },
-		.gate = { CGU_REG_CLKGR1, 24 },
+		.gate = { CGU_REG_CLKGR0, 24 },
 	},
 
 };
