Date: Sun, 6 Jul 2025 05:31:56 +0200
Subject: [PATCH] Register custom C2A board

Upstream-Status: Inappropriate [C2A board support is maintained in this layer]

---
 arch/arm/mach-imx/mx6/Kconfig | 15 +++++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/mach-imx/mx6/Kconfig b/arch/arm/mach-imx/mx6/Kconfig
index 7fd9e295..0b6cad5a 100644
--- a/arch/arm/mach-imx/mx6/Kconfig
+++ b/arch/arm/mach-imx/mx6/Kconfig
@@ -540,6 +540,18 @@ config TARGET_MX6UL_14X14_EVK
 	select FSL_BLOB
 	select ARCH_MISC_INIT
 
+config TARGET_MX6UL_C2A
+	bool "mx6ul_c2a"
+	depends on MX6UL
+	select BOARD_LATE_INIT
+	select DM
+	select DM_THERMAL
+	select SUPPORT_SPL
+	select IMX_MODULE_FUSE
+	select OF_SYSTEM_SETUP
+	imply CMD_DM
+	select ARCH_MISC_INIT
+
 config TARGET_MX6UL_ENGICAM
 	bool "Support Engicam GEAM6UL/Is.IoT"
 	depends on MX6UL
@@ -826,6 +838,7 @@ source "board/freescale/mx6sllevk/Kconfig"
 source "board/freescale/mx6sxsabresd/Kconfig"
 source "board/freescale/mx6sxsabreauto/Kconfig"
 source "board/freescale/mx6ul_14x14_evk/Kconfig"
+source "board/freescale/mx6ul_c2a/Kconfig"
 source "board/freescale/mx6ullevk/Kconfig"
 source "board/bosch/acc/Kconfig"
 source "board/grinn/liteboard/Kconfig"
-- 
2.50.0
