Date: Sat, 5 Jul 2025 22:56:22 +0200
Subject: [PATCH 2/2] Change -Os to more aggressive -Oz for CONFIG_CC_OPTIMIZE_FOR_SIZE

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b58a061cb..6086cdccb 100644
--- a/Makefile
+++ b/Makefile
@@ -809,7 +809,7 @@ ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
 KBUILD_CFLAGS += -O2
 KBUILD_RUSTFLAGS += -Copt-level=2
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
-KBUILD_CFLAGS += -Os
+KBUILD_CFLAGS += -Oz
 KBUILD_RUSTFLAGS += -Copt-level=s
 endif
 
-- 
2.50.0

